integration-tests/redis_test.go

```diff
@@ -78,10 +78,139 @@ var _ = Describe("Redis", Label("Redis"), func() {
+						"plan defined properties cannot be changed",
+					),
+				))
+
+				Expect(mockTerraform.ApplyInvocations()).To(HaveLen(0))
+			},
+			Entry("cache_size", "cache_size", 9),
+			Entry("node_count", "node_count", 5),
+			Entry("redis_version", "redis_version", "4.0"),
+		)
 
-			Expect(err).To(MatchError(ContainSubstring("region: Does not match pattern '^[a-z][a-z0-9-]+$'")))
+		DescribeTable(
+			"should disallow `user_input` properties with the same name as some `computed_input` for clarity",
+			func(prop string, value any) {
+				_, err := broker.Provision(redisServiceName, "small", map[string]any{prop: value})
+
+				Expect(err).To(MatchError(
+					ContainSubstring(
+						"additional properties are not allowed",
+					),
+				))
+
+				Expect(mockTerraform.ApplyInvocations()).To(HaveLen(0))
+			},
+			Entry("labels", "labels", "a-valid-list-of-labels"),
+		)
+
+		DescribeTable("currently doesn't provide validations constraints for some properties",
+			func(params map[string]any) {
+				_, err := broker.Provision(redisServiceName, "small", params)
+				Expect(err).NotTo(HaveOccurred())
+			},
+			Entry(
+				"aws_vpc_id is never validated by the brokerpak logic",
+				map[string]any{"aws_vpc_id": stringOfLen(99)},
+			),
+			Entry(
+				"node_type is never validated by the brokerpak logic",
+				map[string]any{"node_type": stringOfLen(99)},
+			),
+			Entry(
+				"elasticache_subnet_group currently is never validated by the brokerpak logic",
+				map[string]any{"elasticache_subnet_group": stringOfLen(99)},
+			),
+			Entry(
+				"elasticache_vpc_security_group_ids is never validated by the brokerpak logic",
+				map[string]any{"elasticache_vpc_security_group_ids": stringOfLen(99)},
+			),
+		)
+
+		It("should provision a plan", func() {
+			instanceID, err := broker.Provision(redisServiceName, "small", map[string]any{})
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(mockTerraform.FirstTerraformInvocationVars()).To(
+				SatisfyAll(
+					HaveKey("instance_name"),
+					HaveKeyWithValue("labels", HaveKeyWithValue("pcf-instance-id", instanceID)),
+					HaveKeyWithValue("region", "us-west-2"),
+					HaveKeyWithValue("cache_size", BeNumerically("==", 2)),
+					HaveKeyWithValue("node_count", BeNumerically("==", 1)),
+					HaveKeyWithValue("redis_version", "6.0"),
+					HaveKeyWithValue("aws_vpc_id", BeEmpty()),
+					HaveKeyWithValue("node_type", BeEmpty()),
+					HaveKeyWithValue("elasticache_subnet_group", BeEmpty()),
+					HaveKeyWithValue("elasticache_vpc_security_group_ids", BeEmpty()),
+					HaveKeyWithValue("aws_access_key_id", "aws-access-key-id"),
+					HaveKeyWithValue("aws_secret_access_key", "aws-secret-access-key"),
+				))
+		})
+
+		It("should allow properties to be set on provision", func() {
+			_, err := broker.Provision(redisServiceName, "small", map[string]any{
+				"instance_name":                      "some-valid-instance-name",
+				"region":                             "some-valid-region",
```

Comment:
thoughts: I like to use real values to use them as documentation

Reply:
Sounds reasonable.
My reasoning though was to make it explicit that the value in these fields' are not really validated.
But let me know if you prefer me to use "believable" values here (I say "believable" instead of "real" because we don't have guarantees that the same `region` values will existing in the future, and of course we won't paste an aws_secret_access_key here, etc. Believable in the sense that they resemble real values).